rust/libnewsboat/src/configcontainer.rs

```diff
@@ -0,0 +1,1173 @@
+use std::collections::{HashMap, HashSet};
+use std::sync::{Arc, Mutex};
+
+#[derive(Clone, Debug, PartialEq)]
+pub enum ConfigDataType {
+    Bool,
+    Int,
+    Enum,
+    Str,
+    Path,
+}
+
+#[derive(Clone, Debug, PartialEq)]
+pub enum FeedSortMethod {
+    None,
+    FirstTag,
+    Title,
+    ArticleCount,
+    UnreadArticleCount,
+    LastUpdated,
+    LatestUnread,
+}
+
+#[derive(Clone, Debug, PartialEq)]
+pub enum ArtSortMethod {
+    Title,
+    Flags,
+    Author,
+    Link,
+    Guid,
+    Date,
+    Random,
+}
+
+#[derive(Clone, Debug, PartialEq)]
+pub enum SortDirection {
+    Asc,
+    Desc,
+}
+
+#[derive(Clone, Debug, PartialEq)]
+pub enum ConfigHandlerStatus {
+    Success,
+    InvalidCommand,
+    TooFewParams,
+    TooManyParams,
+    InvalidParams(String),
+}
```

Comment:
A more idiomatic approach is to define an `ConfigHandlerError` enum and use `Result<(), ConfigHandlerError>` for the return type. This allows to e.g. use `?` to escape a function early if an error occurs. Is there a reason to define a `Success` constant instead?